scripts/joint_goal_splitter.py

```diff
@@ -129,8 +129,9 @@ def callback(self, goal):
         for i in range(self.number_of_clients):
             self.action_clients[i].send_goal(action_goals[i]), self.feedback_cb
 
+        trajectory_duration = goal.trajectory.points[-1].time_from_start + rospy.Duration(secs=2)
         for i in range(self.number_of_clients):
-            self.action_clients[i].wait_for_result()
+            self.action_clients[i].wait_for_result(timeout=trajectory_duration)
```

Comment:
I haven't tested it, but I believe there would be a few bugs:
1. when you finished waiting for the first action server, the timer starts anew for the 2. one. thus you could wait for trajectory_duration*self.number_of_clients seconds.
2. when waiting for the 1. client to finish, you don't notice immediately, when the 2. fails
3. when something goes wrong the other trajectories should get canceled
4. I'm not sure what happens when wait_for_result times out, it might throw an exception. even if it doesn't, I don't think you can do get_result after a timeout. you should cancel everything, print an error and return fail or something.